src/Core/ExecutablePluginTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Core\RulesPluginTrait.
+ */
+
+namespace Drupal\rules\Core;
+
+use Drupal\Component\Plugin\Exception\ContextException;
+
+/**
+ * Offers common methods for context plugin implementers.
+ */
+trait ExecutablePluginTrait {
+
+  /**
+   * {@inheritdoc}
+   */
+  private function getLabelValue() {
+    $definition = $this->getPluginDefinition();
+
+    if (empty($definition['label'])) {
+      throw new ContextException('The label is not defined.');
```

Comment:
Context exception? This has nothing to do with the context, maybe a plugin exception would be better?